### PR TITLE
[MIG] product_multi_price: Migration to 12.0

### PR DESCRIPTION
**Pull request by chienandalu**
- Minor migration tweaks

cc @Tecnativa TT23610

### CODE REVIEW

File: product_multi_price/tests/test_product_multi_price.py [hunk 1]
```diff
@@ -35,11 +35,13 @@ def setUpClass(cls):
...
+                'value_ids': [(6, 0,
+                    [cls.prod_attr1_v1.id, cls.prod_attr1_v2.id])],
             })]
         })
-        cls.prod_prod_2_1 = product_obj.create({
-            'product_tmpl_id': cls.prod_2.id,
-            'attribute_value_ids': [(6, 0, [cls.prod_attr1_v1.id])],
+        cls.prod_prod_2_1 = cls.prod_2.product_variant_ids[0]
+        cls.prod_prod_2_2 =  cls.prod_2.product_variant_ids[1]
+        cls.prod_prod_2_1.update({
```
Comment (pedrobaeza) [inline product_multi_price/tests/test_product_multi_price.py]:
```suggestion
        cls.prod_prod_2_1.write({
```

File: product_multi_price/tests/test_product_multi_price.py [hunk 2]
```diff
@@ -51,9 +53,7 @@ def setUpClass(cls):
                 }),
             ],
         })
-        cls.prod_prod_2_2 = product_obj.create({
-            'product_tmpl_id': cls.prod_2.id,
-            'attribute_value_ids': [(6, 0, [cls.prod_attr1_v2.id])],
+        cls.prod_prod_2_2.update({
```
Comment (pedrobaeza) [inline product_multi_price/tests/test_product_multi_price.py]:
```suggestion
        cls.prod_prod_2_2.write({
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
And please squash administrative commits in the closest upper one

**Comment by chienandalu** [comment]
@pedrobaeza Changes done

**Review by pedrobaeza — APPROVED** [review]
Supposing Travis is green

**Review by ernestotejeda — COMMENTED** [review]
Thanks David.
Just a little issue:

Peek 2020-06-29 08-33

**Comment by chienandalu** [comment]
> Just a little issue:

Looks like this issue is in v11 as well and comes from some incompatibility of the selection widget for the multi price many2one and the pricelist item auto generated for the pricelist when the pricelist is in state new.

Any ideas @Yajo @Tardo @pedrobaeza ?

**Comment by pedrobaeza** [comment]
selection widget for m2o is not working in tree views (I thought that was modified in v11 already...). Change that widget.

**Comment by chienandalu** [comment]
> selection widget for m2o is not working in tree views

It's on a form and it works fine once the pricelist is saved.

File: product_multi_price/views/product_pricelist_views.xml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="product_pricelist_item_form_view" model="ir.ui.view">
+        <field name="model">product.pricelist.item</field>
+        <field name="inherit_id" ref="product.product_pricelist_item_form_view"/>
+        <field name="arch" type="xml">
+            <field name="base" position="after">
+                <field name="multi_price_name" colspan="6" attrs="{'invisible': [('base', '!=', 'multi_price')], 'required': [('base', '=', 'multi_price')]}" widget="selection"/>
```
Comment (yajo) [inline product_multi_price/views/product_pricelist_views.xml]: As @pedrobaeza said, it seems [you cannot use the selection widget in a m2o field](https://github.com/odoo/odoo/blob/7ea6798fb8b8b0249f56cf4bc5cab627438cd636/addons/web/static/src/js/fields/relational_fields.js#L2352):

```suggestion
                <field name="multi_price_name" colspan="6" attrs="{'invisible': [('base', '!=', 'multi_price')], 'required': [('base', '=', 'multi_price')]}" />
```

**Comment by chienandalu** [comment]
Ok. Changes done

File: product_multi_price/models/product_template.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    def _get_multiprice_pricelist_price(self, rule):
+        if len(self.product_variant_ids) == 1:
+            return (
+                self.product_variant_ids._get_multiprice_pricelist_price(rule))
+        return 0
+
+    @api.model
+    def create(self, vals):
+        product = super().create(vals)
+        if vals.get('price_ids'):
```
Comment (Tardo) [inline product_multi_price/models/product_template.py:40]: Why need rewrite this field?

Comment (pedrobaeza) [reply product_multi_price/models/product_template.py:40]: Clarified in a docstring in the code:
```
Overwrite creation for rewriting the prices (if set and having only
one variant), after the variant creation, that is performed in super.
```

**Review by Tardo — APPROVED** [review]

**Pull request merged**